### PR TITLE
Fixes #7649: Add support for the SMF system on Solaris for service management

### PR DESCRIPTION
**Pull request by jooooooon**
https://www.rudder-project.org/redmine/issues/7649

### CODE REVIEW

File: tests/style/tests/all_shell_test_should_be_set_e.sh [hunk 1]
```diff
@@ -10,7 +10,7 @@ ALL_SHELL_TESTS=`find "${NCF_TREE}/../tests/style/" -name "*.sh"`
 ERRORS=0
 for file in ${ALL_SHELL_TESTS}
 do
-  if ! egrep "^\s*set\s+-e[\s#]*$" ${file} > /dev/null; then
+  if ! egrep "^[ 	]*set[ 	]+-e" ${file} > /dev/null; then
```
Comment (jooooooon) [inline tests/style/tests/all_shell_test_should_be_set_e.sh]: Solaris' version of egrep doesn't know about \\\ \s \\\

**Comment by jooooooon** [comment]
PR updated

File: tree/30_generic_methods/service_action.cf [hunk 2]
```diff
@@ -47,7 +47,17 @@ bundle agent service_action(service_name, action)
       "svc_action_cmd[stop]"      string => "/usr/bin/stopsrc -s ${service_name}";
       "action_command"            string => "${svc_action_cmd[${action}]}";
 
-    !systemctl_utility_present.!service_utility_present.!startsrc_utility_present.init_d_directory_present::
+    !systemctl_utility_present.!service_utility_present.!startsrc_utility_present.svcadm_utility_present::
+      "svc_action_cmd[restart]"   string => "${paths.path[svcadm]} restart -s ${service_name}";
+      "svc_action_cmd[refresh]"   string => "${paths.path[svcadm]} refresh -s ${service_name}";
+      "svc_action_cmd[reload]"    string => "${svc_action_cmd[refresh]}";
+      "svc_action_cmd[enable]"    string => "${paths.path[svcadm]} enable -s ${service_name}";
+      "svc_action_cmd[start]"     string => "${svc_action_cmd[enable]}";
```
Comment (peckpeck) [inline tree/30_generic_methods/service_action.cf]: I think we should add -t option to start and to stop.
If the -t option is specified, svcadm temporarily enables each service instance.
ie this is the difference between start and enable

File: tree/30_generic_methods/service_action.cf [hunk 3]
```diff
@@ -69,18 +79,18 @@ bundle agent service_action(service_name, action)
...
       "class copy"                usebundle => _classes_copy("command_execution_${canonified_action_command}", "${class_prefix}");
 
-    systemctl_utility_present|service_utility_present|init_d_directory_present|startsrc_utility_present|(windows.is_valid_action)::
+    systemctl_utility_present|service_utility_present|init_d_directory_present|startsrc_utility_present|svcadm_utility_present|(windows.is_valid_action)::
 
       "report"                    usebundle => _logger("Run action ${action} on service ${service_name}", "${class_prefix}"),
                                  ifvarclass => "${class_prefix}_reached";
 
-    (!systemctl_utility_present.!service_utility_present.!init_d_directory_present.!startsrc_utility_present.!windows)|(windows.!is_valid_action)::
+    (!systemctl_utility_present.!service_utility_present.!init_d_directory_present.!startsrc_utility_present.!svcadm_utility_present!windows)|(windows.!is_valid_action)::
```
Comment (peckpeck) [inline tree/30_generic_methods/service_action.cf]: There is a dot missing after svcadm

File: tree/30_generic_methods/service_check_started_at_boot.cf [hunk 4]
```diff
@@ -38,7 +38,10 @@ bundle agent service_check_started_at_boot(service_name)
     pass1.!systemctl_utility_present.!chkconfig_utility_present.lsitab_utility_present::
       "command_to_check"   string => "/usr/sbin/lsitab -a | egrep '^${service_name}:[0-9]+:(respawn|boot|bootwait|wait|once|initdefault|sysinit):'";
 
-    pass1.(!(systemctl_utility_present|chkconfig_utility_present|lsitab_utility_present)|broken_systemctl)::
+    pass1.!systemctl_utility_present.!chkconfig_utility_present.!lsitab_utility_present.svcs_utility_present::
+      "command_to_check"   string => "${paths.path[svcs]} -l ${service_name} | egrep '^enabled [ 	]+(true|false \(temporary\))'";
+
+    pass1.(!(systemctl_utility_present|chkconfig_utility_present|lsitab_utility_present|svcs_utility_present)|broken_systemctl)::
```
Comment (peckpeck) [inline tree/30_generic_methods/service_check_started_at_boot.cf]: Could you rewrite this to use the same convention as above (!x.!y instead of !(x|y) )

**Comment by peckpeck** [comment]
Superseded by https://github.com/Normation/ncf/pull/296

**Pull request closed without merging**